Production_scripts/Process_extracts/extract_gp_ooh_consultations.R

```diff
@@ -0,0 +1,329 @@
+  # Match to cost lookup
+  left_join(ooh_cost_lookup, by = c("hbtreatcode", "year")) %>%
+  rename(
+    cost_total_net = cost_per_consultation
+  ) %>%
+  create_day_episode_costs(record_keydate1, cost_total_net)
+
+
+# Data Cleaning--------------------------------------
+
+ooh_clean <- ooh_costs %>%
+  # rename outcomes
+  rename(
+    ooh_outcome.1 = outcome.1,
+    ooh_outcome.2 = outcome.2,
+    ooh_outcome.3 = outcome.3,
+    ooh_outcome.4 = outcome.4
+  ) %>%
+  mutate(
+    # Replace location unknown with blank. Should this be NA?
+    location = if_else(location == "UNKNOWN", "", location),
+    recid = "OoH",
+    smrtype = case_when(
+      smrtype == "DISTRICT NURSE" ~ "OOH-DN",
+      smrtype == "DOCTOR ADVICE/NURSE ADVICE" ~ "OOH-Advice",
+      smrtype == "HOME VISIT" ~ "OOH-HomeV",
+      smrtype == "NHS 24 NURSE ADVICE" ~ "OOH-NHS24",
+      smrtype == "PCEC/PCC" ~ "OOH-PCC",
+      TRUE ~ "OOH-Other"
+    ),
+    kis_accessed = case_when(
+      kis_accessed == "Y" ~ 1,
+      kis_accessed == "N" ~ 0,
+      TRUE ~ 9
+    )
+  ) %>%
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # split time from date
+  mutate(
+    key_time1 = as_hms(substr(record_keydate1, 12, 19)),
+    key_time2 = as_hms(substr(record_keydate2, 12, 19)),
+    record_keydate1 = substr(record_keydate1, 1, 10),
+    record_keydate2 = substr(record_keydate2, 1, 10)
+  )
+
+# Keep the location descriptions as a lookup.
+location_lookup <- ooh_clean %>%
+  group_by(location) %>%
+  summarise(
+    location_description = first(location_description)
+  ) %>%
+  ungroup()
+
+ooh_clean <- ooh_clean %>%
+  arrange(guid, chi) %>%
+  # group for getting row order
+  group_by(guid, chi) %>%
+  mutate(
+    row_order = row_number(),
+    ooh_cc = 0,
+    ooh_cc = case_when(
+      ooh_cc == 0 & row_order == 1 | chi != lag(chi, default = first(chi)) ~ 1,
+      ooh_cc == 0 & guid != lag(guid) ~ lag(ooh_cc, default = first(ooh_cc)) + 1,
+      ooh_cc == 0 ~ lag(ooh_cc, default = first(ooh_cc))
+    )
+  )
+
+
+## Save Outfile -------------------------------------
+
+outfile <- ooh_clean %>%
+  arrange(
+    chi,
+    record_keydate1,
+    keytime1
+  )
+select(
+  year,
+  recid,
+  smrtype,
+  record_keydate1,
+  record_keydate2,
+  keytime1,
+  keytime2,
+  chi,
+  gender,
+  dob,
+  age,
+  gpprac,
+  postcode,
+  hbrescode,
+  datazone,
+  hscp,
+  hbtreatcode,
+  location,
+  attendance_status,
+  kis_Accessed,
+  refsource,
+  contains("diag"),
+  contains("ooh_outcome"),
+  cost_total_net,
+  apr_cost,
+  may_cost,
+  jun_cost,
+  jul_cost,
+  aug_cost,
+  sep_cost,
+  oct_cost,
+  nov_cost,
+  dec_cost,
+  jan_cost,
+  feb_cost,
+  mar_cost,
+  ooh_CC
+)
+
+## Housekeeping -------------------------------------
+
+# Delete TEMP files
+# diagnosis
+file.remove(paste0(
+  get_year_dir(year = year),
+  "/gp-diagnosis-data-20",
+  year, ".rds"
+))
+
+# outcomes
+file.remove(paste0(
+  get_year_dir(year = year),
+  "/gp-outcomes-data-20",
+  year, ".rds"
+))
+
```

Comment:
I know in one of mine I removed the deletion of temporary files as in R you do not need to save these out, so unsure if required or not?